controllers/csm_controller.go

```diff
@@ -1524,12 +1529,11 @@ func (r *ContainerStorageModuleReconciler) GetK8sClient() kubernetes.Interface {
 	return r.K8sClient
 }
 
-func (r *ContainerStorageModuleReconciler) GetMatchingNodes(ctx context.Context, labelKey string, labelValue string) (*corev1.NodeList, error) {
-	nodeList := &corev1.NodeList{}
-	opts := []client.ListOption{
-		client.MatchingLabels{labelKey: labelValue},
+func (r *ContainerStorageModuleReconciler) ZoneValidation(ctx context.Context, cr *csmv1.ContainerStorageModule) error {
+	err := drivers.ValidateZones(ctx, cr, r.Client)
+	if err != nil {
+		return fmt.Errorf("ZoneValidation failed with error: %v", err)
```

Comment:
I know the linter usually complains about error messages starting with capital letters

Reply:
It could also be more informative, like "array zone validation failed: %v". We should pay more attention to usefulness of log messages while keeping them concise.

Reply:
Yep, probably didn't catch this because it matches the function name. re-wrote the error message to be more user-friendly.